File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/AssignmentEvaluationColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AssignmentEvaluationRow } from "./AssignmentEvaluationRow";

export interface AssignmentEvaluationColumns {
    Id: Column<AssignmentEvaluationRow>;
    AssignmentTitle: Column<AssignmentEvaluationRow>;
    EvalutionCriteria: Column<AssignmentEvaluationRow>;
    ClassTitle: Column<AssignmentEvaluationRow>;
    SubjectTitle: Column<AssignmentEvaluationRow>;
    TopicTitle: Column<AssignmentEvaluationRow>;
    BloomsIndexCoginitiveSkill: Column<AssignmentEvaluationRow>;
    BloomsWeightage: Column<AssignmentEvaluationRow>;
    InsertDate: Column<AssignmentEvaluationRow>;
    InsertUserId: Column<AssignmentEvaluationRow>;
    UpdateDate: Column<AssignmentEvaluationRow>;
    UpdateUserId: Column<AssignmentEvaluationRow>;
    IsActive: Column<AssignmentEvaluationRow>;
}

export class AssignmentEvaluationColumns extends ColumnsBase<AssignmentEvaluationRow> {
    static readonly columnsKey = 'Exams.AssignmentEvaluation';
    static readonly Fields = fieldsProxy<AssignmentEvaluationColumns>();
}

File: GXpert/GXpert.Web/Modules/Exams/AssignmentEvaluation/AssignmentEvaluationColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Exams.Columns;

[ColumnsScript("Exams.AssignmentEvaluation")]
[BasedOnRow(typeof(AssignmentEvaluationRow), CheckNames = true)]
public class AssignmentEvaluationColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string AssignmentTitle { get; set; }
    [EditLink]
    public string EvalutionCriteria { get; set; }
    public string ClassTitle { get; set; }
    public string SubjectTitle { get; set; }
    public string TopicTitle { get; set; }
    public string BloomsIndexCoginitiveSkill { get; set; }
    public float BloomsWeightage { get; set; }
}